mcm/tests/campaign/test_campaign.py

```diff
@@ -0,0 +1,268 @@
+"""
+This module tests the API operations related with
+the campaign entity.
+"""
+
+import uuid
+from pathlib import Path
+from copy import deepcopy
+from importlib.resources import files
+import pytest
+import tests.campaign as campaign
+from tests.base_test_tools import Entity, EntityTest, config
+from tests.api_tools import McM, Roles
+
+
+class CampaignAPI(Entity):
+    """
+    Implements a simple test API client focused
+    only on the campaign operations.
+    """
+
+    def __init__(self, mcm: McM) -> None:
+        mock_path: Path = files(campaign).joinpath("static/campaign.json")
+        super().__init__(mock_path, mcm)
+        self.object_type: str = "campaigns"
+
+    def mockup(self):
+        mock = deepcopy(self._original_mockup)
+        suffix: str = str(uuid.uuid4()).split("-")[0]
+        new_id: str = f"Test{mock.get('prepid', '')}R{suffix}"
+        mock["_id"] = new_id
+        mock["prepid"] = new_id
+        return mock
+
+    def create(self, mockup: dict):
+        return self.mcm.put(self.object_type, mockup)
+
+    def retrieve(self, mockup: dict):
+        return self.mcm.get(self.object_type, mockup.get("prepid"))
+
+    def update(self, mockup: dict):
+        return self.mcm.update(self.object_type, mockup)
+
+    def delete(self, mockup: dict):
+        return self.mcm.delete(self.object_type, mockup.get("prepid"))
```

Comment:
I understand your point of view, but maybe future entities like `chain_campaign` and `chain_request` would require more complex procedures, so let's leave the child class to implement it. In the end, we can give it a refactor in case it is required.